pkg/jx/cmd/create_vault.go

```diff
@@ -286,3 +365,18 @@ func (o *CreateVaultOptions) exposeVault(vaultService string) error {
 	}
 	return upgradeIngOpts.Run()
 }
+
+// CreateAuthServiceAccount creates a Serivce Account for the Auth service for vault
+func CreateAuthServiceAccount(client kubernetes.Interface, vaultName, namespace, clusterName string) (string, error) {
+	serviceAccountName := AuthServiceAccountName(vaultName)
+	_, err := serviceaccount.CreateServiceAccount(client, namespace, serviceAccountName)
+	if err != nil {
+		return "", errors.Wrap(err, "creating vault auth service account")
+	}
+	return serviceAccountName, nil
+}
+
+// AuthServiceAccountName creates a service account name for a given vault and cluster name
```

Comment:
What is the reason to move this function here? I think that it is also used by other commands like `delete vault`.

Reply:
Also I think the name was truncated to avoid exceeding the allowed resource name length.

Reply:
It was in a GKE specific place before while it will probably be used for all cloud providers. Also it is only used in create_vault. Maybe it should be used in `delete vault`, but it isn't now. So I'll let it be for now.